nil-docs/zkllvm/getting-started/installation.mdx

```diff
@@ -0,0 +1,259 @@
+    ```
+  </TabItem>
+</Tabs>
+
+### Build the Rust compiler
+
+There are two possible ways to install and build the Rust compiler.
+
+- Use a pre-binary installation
+- Build the compiler from source
+
+#### Using a pre-built binary
+
+If `rustup` is already installed, it is possible to install the Rust compiler from a pre-built binary which is easier and faster compared to building it from source.
+
+Install `rslang` by executing the following command.
+
+```bash
+curl --proto '=https' --tlsv1.2 -sSf https://cdn.jsdelivr.net/gh/NilFoundation/zkllvm@master/rslang-installer.py | python - --channel nightly
+```
+
+Alternatively, download `rslang-installer.py` and run it.
+
+```bash
+curl --proto '=https' --tlsv1.2 -O https://cdn.jsdelivr.net/gh/NilFoundation/zkllvm@master/rslang-installer.py
+python rslang-installer.py --channel nightly
+```
+
+:::info[Standalone installation]
+
+To install `rslang` without `rustup`, run the below command.
+
+```bash
+curl --proto '=https' --tlsv1.2 -O https://cdn.jsdelivr.net/gh/NilFoundation/zkllvm@master/rslang-installer.py
+python rslang-installer.py --no-rustup --prefix PATH
+```
+
+:::
+
+Regardless of the chosen installation method, the Rust compiler should now be accessible as follows.
+
+```bash
+rustc +zkllvm -V
+```
+
+#### Build the compiler from source
+
+The first step to building the Rust compiler is adding Rust tools to `cmake`.
+
+<Tabs groupId='buildSystem'>
+  <TabItem value='unix' label='Unix'>
+    ```bash
+    cmake -G "Unix Makefiles" -B ${ZKLLVM_BUILD:-build} -DCMAKE_BUILD_TYPE=Release -DRSLANG_BUILD_EXTENDED=TRUE -DRSLANG_BUILD_TOOLS=cargo .
+    ```
+  </TabItem>
+  <TabItem value='ninja' label='Ninja'>
+    ```bash
+    cmake -G "Ninja" -B ${ZKLLVM_BUILD:-build} -DCMAKE_BUILD_TYPE=Release -DRSLANG_BUILD_EXTENDED=TRUE -DRSLANG_BUILD_TOOLS=cargo .
+    ```
+  </TabItem>
+</Tabs>
+
+Afterward, export the path containing LLVM libraries.
+
+```bash
+export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$(pwd)/build/libs/circifier/llvm/lib"
+```
+
+Use the below command to build `rustc` and `cargo`.
+
+<Tabs groupId='buildSystem'>
+  <TabItem value='unix' label='Unix'>
+    ```bash
+    make -C ${ZKLLVM_BUILD:-build} rslang -j$(nproc)
+    ```
+  </TabItem>
+  <TabItem value='ninja' label='Ninja'>
+    ```bash
+    ninja -C ${ZKLLVM_BUILD:-build} rslang -j$(nproc)
+    ```
+  </TabItem>
+</Tabs>
+
+After the build completes, export the `RSLANG` environment variable.
+
+```bash
+export RSLANG="$(pwd)/build/libs/rslang/build/host"
+```
+
+Verify the versions of `rustc` and `cargo` by running the following commands.
+
+```bash
+RUSTC=$RSLANG/stage1/bin/rustc $RSLANG/stage1-tools-bin/cargo --version
+```
+
+:::tip[Finding `rustc` and `cargo`]
+
+If the terminal cannot find `rustc` and `cargo`, run `find ~/zkllvm -name rustc` and `find ~/zkllvm -name cargo`.
+
+:::
+
+:::info[Advanced compiling with Rust]
+
+To use the default build system supplied by Rust, build `zkllvm` first.
```

Comment:
I think, this may sound a bit misleading. "default build system supplied by Rust" - it's not about Cargo, it's about `bootstrap` (`x.py`). That's the build system for Rust compiler and tools, not for Rust code. I would probably paraphrase: "To use original Rust bootstrapping system, ..."

Reply:
thank you, addressed